qiskit_experiments/calibration_management/calibration_utils.py

```diff
@@ -12,51 +12,138 @@
+        schedule: The schedule for which to get the parameters in the channels.
 
     Returns:
-        True if ``schedule``calls a ``ScheduleBlock`` with name ``schedule_name``.
+        The set of parameters explicitly defined in the schedule.
+
+    Raises:
+        CalibrationError: If a channel is parameterized by more than one parameter.
+        CalibrationError: If the parameterized channel index is not formatted properly.
     """
-    blocks_have_schedule = False
-
-    for block in schedule.blocks:
-        if isinstance(block, Call):
-            if block.subroutine.name == schedule_name:
-                return True
-            else:
-                blocks_have_schedule = blocks_have_schedule or _used_in_calls(
-                    schedule_name, block.subroutine
+
+    # The channel indices need to be parameterized following this regex.
+    __channel_pattern__ = r"^ch\d[\.\d]*\${0,1}[\d]*$"
+
+    param_indices = set()
+
+    # Schedules with references do not explicitly have channels. This needs special handling.
+    if schedule.is_referenced():
+        for block in schedule.blocks:
+            if isinstance(block, ScheduleBlock):
+                param_indices.update(validate_channels(block))
+
+        return param_indices
+
+    regex = re.compile(__channel_pattern__)
+    for ch in schedule.channels:
+        if isinstance(ch.index, ParameterExpression):
+            if len(ch.index.parameters) != 1:
+                raise CalibrationError(f"Channel {ch} can only have one parameter.")
+
+            param_indices.add(ch.index)
+            if regex.match(ch.index.name) is None:
+                raise CalibrationError(
+                    f"Parameterized channel must correspond to {__channel_pattern__}"
                 )
 
-        if isinstance(block, ScheduleBlock):
-            blocks_have_schedule = blocks_have_schedule or _used_in_calls(schedule_name, block)
+    return param_indices
+
+
+def reference_info(
+    reference: Tuple[str, ...],
+    qubits: Optional[Tuple[int, ...]] = None,
+) -> Tuple[str, Tuple[int, ...]]:
+    """Extract reference information from the reference tuple.
+
+    Args:
+        reference: The reference of a Reference instruction in a ScheduleBlock.
+        qubits: Optional argument to reorder the references.
+
+    Returns:
+        A string corresponding to the name of the referenced schedule and the qubits that
+        this schedule applies to.
+
+    Raises:
+        CalibrationError: If ``reference`` is not a tuple.
+        CalibrationError: If ``reference`` is not a tuple of reference name and the qubits that
+            that the schedule applies to.
+    """
+    if not isinstance(reference, tuple):
+        raise CalibrationError(f"A schedule reference must be a tuple. Found {reference}.")
+
+    ref_schedule_name, ref_qubits = reference[0], reference[1:]
+
+    # Convert a single-qubit reference to a tuple type.
+    if isinstance(ref_qubits, str):
```

Comment:
`ref_qubits` will always be a tuple because a slice of a tuple is always a tuple.

Reply:
Thanks. Removed this part of code here: [fba863c](https://github.com/Qiskit/qiskit-experiments/pull/889/commits/fba863c89c98508652a7d383733e7d7bfbe758aa)